### PR TITLE
Integration tests: logging is shut down when CLI finished

### PR DESCRIPTION
**Pull request by rvl**
Relates to #851.

# Overview

Adds integration test case for clean logging shutdown.

### CODE REVIEW

File: lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs [hunk 1]
```diff
@@ -159,6 +159,20 @@ spec = do
...
+                    ["serve"
+                    , "--node-port"
+                    , show (ctx ^. typed @(Port "node"))
+                    , "--random-port"
+                    , "--verbose"
+                    , "--genesis-block-hash"
+                    , block0H
+                    ]
+            let process = proc' (commandName @t) args
+            (out, _) <- collectStreams (10, 0) process
```
Comment (jonathanknowles) [inline lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs]: Hi @rvl. Are you able to say a little bit about where `10` and `0` came from here?

**Review by KtorZ — APPROVED** [review]

**Review by piotr-iohk — APPROVED** [review]

File: lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs (hunk 1 above)
Comment (rvl) [reply lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs]: These are the number of output lines to collect before finishing the test.
The `collectStreams` function is a bit fishy actually.

**Review by paweljakubas — APPROVED** [review]

**Comment by paweljakubas** [comment]
rebased against master

**Comment by paweljakubas** [comment]
interesting : https://hydra.iohk.io/build/1303898/nixlog/31

**Comment by rvl** [comment]
@paweljakubas 

> interesting : https://hydra.iohk.io/build/1303898/nixlog/31

This is a similar/the same error #931 that occurred on Buildkite and caused the [Bors build to fail](https://buildkite.com/input-output-hk/cardano-wallet/builds/3478).

**Comment by rvl** [comment]
Will try again :crossed_fingers: 

bors r+

**Pull request merged**